heartbeat/galera

```diff
@@ -492,6 +514,19 @@ detect_first_master()
     done
 
     for node in $nodes_recovered $nodes; do
+        safe_to_bootstrap=$(get_safe_to_bootstrap $node)
+
+        if [ "$safe_to_bootstrap" = "1" ]; then
+            # Galera marked the node as safe to boostrap during shutdown. Let's just
+            # pick it as our bootstrap node.
```

Comment:
Does Galera marking nodes safe to bootstrap mean they are at the latest version or just that they shutdown cleanly. If it's the latter, then this isn't sufficient. I suspect it's the former but just checking/curious.

Reply:
"safe_to_bootstrap" means it's was the last one leaving the cluster. So it was cleanly shutdown and has the latest data.